### DISCUSSION TITLE
Cannot set a custom shell in profiles

### DISCUSSION THREAD

**Discussion by kuesji** [netblue30/firejail#5208] [Q&A] [answered]
i was trying to run a program from profile file directly instead of supplying program name via command line.
my profile file is:
```
[kuesji@kpc desktop]$ cat ~/.config/firejail/hello.profile
private
shell /bin/bash
```

and output of command is:
```
[kuesji@kpc desktop]$ firejail hello
Reading profile /data/home/kuesji/.config/firejail/hello.profile
Error: line 2 in /data/home/kuesji/.config/firejail/hello.profile is invalid
```

is `shell program-name` doesn't work at profile file or i'm doing something wrong?

**Comment by rusty-snake** [comment, marked as answer]
The manpage (`firejail-profile(5)`) documents only `shell none`. So either the documentation is missing something or `shell SHELL` isn't supported. If we look in the source, we find that there's only

https://github.com/netblue30/firejail/blob/756fed6bd3a5aae5ca42f4268eca9b7146fbd4f4/src/firejail/profile.c#L369

and no other `strcmp(ptr, "shell ")`.

Now the question arises whether this is intentionally not implemented or just forgotten (because it's less useful).

Additional: Why do you want `shell /bin/bash`? What are you trying to solve?

**Comment by kuesji** [comment]
hmm, i guess it's not a useful case for firejail and it didn't implemented. 
also `shell /bin/bash` is just example. you can put something else to here.

i tell my use case. i am using multiple sandbox with same app (  like firefox ).
i thought i can add firefox to end of profile file as `shell /usr/bin/firefox` and run sandbox by just writing `firejail firefox-yt` or `firejail firefox-work` instead of `firejail --profile=my_profile_path firefox`.

i didn't saw something like exec and i thought i achieve same thing by using shell.

**Reply by rusty-snake** [reply]
> i thought i achieve same thing by using shell.

AFAIK the shell is used like `$SHELL -c program-and-argument` so you would end-up with `firefox -c xyz`

> firejail --profile=my_profile_path firefox

1. You can write `firejail --profile=firefox-yt /usr/bin/firefox`
2. You can add aliases to your bashrc.

**Reply by kuesji** [reply]
i guess i'm going with third way and write a script to parse program name from end of profile. 
btw, thanks you for your response.

**Reply by rusty-snake** [reply]
You make your life complex. Instead of a script that parses something just create a script like

```sh
#!/bin/sh
exec firejail --profile=firefox-work /usr/bin/firefox
```

or add an alias for it.

**Reply by kuesji** [reply]
it is done already. so, no need to worry about this.
